### PR TITLE
feat: upload chunk for readiness check

### PR DESCRIPTION
**Pull request by Cafe137**

### CODE REVIEW

File: src/readiness.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Bee, Utils } from '@ethersphere/bee-js'
+import { logger } from './logger'
+import { StampsManager } from './stamps'
+
+export async function tryUploadingSingleChunk(stampsManager: StampsManager): Promise<boolean> {
+  const chunk = makeChunk()
+  const bee = new Bee(process.env.BEE_API_URL || 'http://localhost:1633')
+  try {
+    await bee.uploadChunk(stampsManager.postageStamp, chunk, { timeout: 3_000, deferred: false })
```
Comment (agazso) [inline src/readiness.ts:9]: Why is this `deferred: false` (or with other words direct upload)?

In the readiness check the goal is to test if the node is working correctly, not that the network is working correctly which may depend on other nodes on the network.

**Comment by AuHau** [comment]
@Cafe137 if I understand correctly, you are doing two things in this PR. You have a fix for the identity header and then the feature of uploading the readiness chunk, right? I assume they are not really related? If so, can you please split them, into separate PRs?

File: src/config.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 export interface AppConfig {
   beeApiUrl: string
-  beeDebugApiUrl?: string
+  beeDebugApiUrl: string
```
Comment (Cafe137) [inline src/config.ts:3]: // TODO discuss this as this is a new requirement to always have debug API

File: src/server.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
-import { BeeDebug } from '@ethersphere/bee-js'
+import { Bee, BeeDebug } from '@ethersphere/bee-js'
 import express, { Application } from 'express'
 import { createProxyMiddleware, Options } from 'http-proxy-middleware'
-import * as bzzLink from './bzz-link'
-import { AppConfig, DEFAULT_HOSTNAME } from './config'
 import { fetchBeeIdentity, getHashedIdentity, HASHED_IDENTITY_HEADER } from './identity'
+import { AppConfig, DEFAULT_HOSTNAME } from './config'
+import * as bzzLink from './bzz-link'
```
Comment (Cafe137) [inline src/server.ts:6]: Revert import sorting

File: test/server.spec.ts [hunk 4]
```diff
@@ -1,20 +1,22 @@
-import { createApp, POST_PROXY_ENDPOINTS } from '../src/server'
-import request from 'supertest'
 import { Bee } from '@ethersphere/bee-js'
 import type { Server } from 'http'
+import request from 'supertest'
+import { DEFAULT_BEE_DEBUG_API_URL } from '../src/config'
+import { createApp, POST_PROXY_ENDPOINTS } from '../src/server'
```
Comment (Cafe137) [inline test/server.spec.ts:5]: Revert import ordering

File: src/readiness.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import { Bee, BeeDebug, Utils } from '@ethersphere/bee-js'
+import { READINESS_TIMEOUT } from './config'
+import { logger } from './logger'
+import { StampsManager } from './stamps'
+
+export async function checkReadiness(bee: Bee, beeDebug: BeeDebug, stampManager?: StampsManager): Promise<boolean> {
```
Comment (AuHau) [inline src/readiness.ts:6]: Could you please add test for this function? You can for example use the https://github.com/ffMathy/FluffySpoon.JavaScript.Testing.Faking (we have it already in devDependencies) to mock the classes and you can do simple unit tests with it.

File: src/readiness.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+    return true
+  } catch (error) {
+    logger.error('unable to upload chunk to bee', error)
+
+    return false
+  }
+}
+
+function makeChunk(seed = '', length = 4096): Uint8Array {
```
Comment (AuHau) [inline src/readiness.ts:36]: If you want to support variable chunk length, shouldn't there be a guard against exceeding the `4096` as max length? As we do not support bigger chunks?

File: src/readiness.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  }
+}
+
+async function tryUploadingSingleChunk(bee: Bee, stampsManager: StampsManager): Promise<boolean> {
+  const chunk = makeChunk()
+  try {
+    await bee.uploadChunk(stampsManager.postageStamp, chunk, { timeout: READINESS_TIMEOUT, deferred: true })
+
+    return true
+  } catch (error) {
```
Comment (AuHau) [inline src/readiness.ts:29]: Are there some cases when this could fail and still be in a "valid state"?

**Review by AuHau — COMMENTED** [review]
Generally LGTM, but would prefer to have some tests for the `checkReadiness()`.

File: src/readiness.ts (hunk 7 above)
Comment (Cafe137) [reply src/readiness.ts:29]: Not really, no. If we do not have postage stamp management, then it just tests the health endpoint. We can change that later to readiness endpoint on Bee as well.

**Review by AuHau — APPROVED** [review]
Awesome! Good job!
Thanks for all the tweaks ;-)

**Pull request merged**